0000-experiment-dataframe.md

```diff
@@ -0,0 +1,320 @@
+
+_Pain points_:
+
+Currently the experiment outcomes are `List[AnalysisResult]` and the query through the REST API is created per single result. This user suffers from the overhead of saving and loading the results from the remote database via the experiment service.
+
+_Solutions_:
+
+This user wants to use data frames to consolidate and simplify data exchange.
+
+#### User2
+- (story1) User wants to do analysis of many experiments data taken over time. Such analysis may include, for example, [TLS landscape visualization](https://www.nature.com/articles/s41534-022-00643-y/figures/4), in which multiple relaxation spectroscopy (freq. v.s. P1) curves are arranged in line with time.
+- (story2) User wants to get multiple curve data subsets from batched `CurveAnalysis` data and visualize them with non-QE plotter, such as [seaborn](https://seaborn.pydata.org/). For example, one may run `T1` experiment with multiple qubits in parallel, and compare the decay curves of different qubits in a single figure.
+
+_Pain points_:
+
+Currently the curve data visualization is tied to the protected `CurveAnalysis._run_analysis` method and we cannot easily access to the curve data. We are lucking the portable data format for curve data, i.e. (x, y, y_err).
+
+_Solutions_:
+
+This user wants to use data frames to make the curve data portable, and want to store it in `ExperimentData.artifacts`.
+
+#### User3
+
+User wants to easily perform custom analysis on outcomes of experiments. For example, the [probabilistic error cancellation](https://arxiv.org/abs/2201.09866) first estimates fidelities of various Pauli channels with RB family experiment, then an experimentalist will develop a noise model tailored to a quantum circuit with these fidelities.
+
+_Pain points_:
+
+Currently analysis results are retuned as a list of `AnalysisResult` and they are not well organized. The experimentalist needs to iterate over all list items and check the entry name and extra metadata to filter the result of interest.
+
+_Solutions_:
+
+This user wants to use data frame to organize the analysis results so it is easy to work with.
+
+
+## Design Proposal
+
+In this proposal, we are going to replace the data format of analysis results, based off of our users survey.
+
+### A. Experiment data
+
+We propose to replace `ExperimentData._analysis_results` data format from `ThreadSafeOrderedDict` to the thread safe wrapper of the pandas data frame. This data frame may consists of following columns:
+
+- name (str): Same as `AnalysisResultData.result_type`. This name will appear in the result database.
+- value (Any): Same as `AnalysisResultData.result_data["_value"]`. Any value that QE analysis will generate. This value must be serializable through the `ExperimentEncoder`.
+- chisq (float): Same as `AnalysisResultData.chisq`. Chi-squared value of the entry. Probably this can be moved to extra dict.
```

Comment:
Agree this is probably not necessary as a main column as it is experiment specific.